spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -51,10 +56,50 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
     this.get(columns, ViewFormat.withName(format))
   }
 
-  override  def getRandomSampleByNum(numSamples: Int, withReplacement: Boolean,
+  override  def getRandomSampleByNum(numSamples: Long, withReplacement: Boolean,
     seed:
   Int): DDF = {
-    null.asInstanceOf[DDF]
+
+    if (numSamples < 0) {
+      throw new IllegalArgumentException("Number of samples must be larger than or equal to 0")
+    }
+
+    if (!withReplacement){
+      mDDF.getSqlHandler.sql2ddf(s"select * from ${mDDF.getSchema.getTableName} order by rand($seed) limit $numSamples")
+    } else {
+      // First, create a Spark DF that contain random numbers from 0 to num_rows - 1
+      val numRows = mDDF.getNumRows
+      val sqlContext = mDDF.getManager.asInstanceOf[SparkDDFManager].getHiveContext
+      val sc = sqlContext.sparkContext
+      val uniRDD = uniformRDD(sc, numSamples, 10, seed).map(x => (((numRows - 1) * x).toLong, ((numRows - 1) * x).toLong))
+
+      // Then join it with the DDF
+      val rddRow: RDD[Row] = mDDF.asInstanceOf[SparkDDF].getRDD(classOf[Row])
+      val joinedRDD = uniRDD.leftOuterJoin(rddRow.zipWithIndex().map(x => (x._2, x._1)))
+
+      // Get an RDD of rows, create a new SparkDF and then a DDF
+      val sampledRDD: RDD[Row] = joinedRDD.map(x => x._2._2.get)
+      val df: DataFrame = mDDF.getRepresentationHandler.get(classOf[DataFrame]).asInstanceOf[DataFrame]
+      val sampledDF = sqlContext.createDataFrame(sampledRDD, df.schema)
+
+
+      val manager = this.getManager
+      val schema = SchemaHandler.getSchemaFromDataFrame(sampledDF)
+      schema.setTableName(mDDF.getSchemaHandler.newTableName())
```

Comment:
you don't need to do this, also we have function newDDFFromSparkDataFrame in SparkDDFManager to create DDF from DataFrame